cmd/worker/main.go

```diff
@@ -65,23 +71,90 @@ func main() {
 		logger.Info("Configuration file not found.")
 	}
 
+	// Temporal client.
+	tc, err := temporalsdk_client.Dial(temporalsdk_client.Options{
+		HostPort:  cfg.Temporal.Address,
+		Namespace: cfg.Temporal.Namespace,
+		Logger:    temporal.Logger(logger.WithName("temporal")),
+	})
+	if err != nil {
+		logger.Error(err, "Unable to create Temporal client.")
+		os.Exit(1)
+	}
+	defer tc.Close()
+
 	ctx, cancel := context.WithCancel(context.Background())
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt)
-	go func() { <-c; cancel() }()
+	var g run.Group
 
-	m := workercmd.NewMain(logger, cfg)
+	// Worker.
+	{
+		done := make(chan struct{})
+		m := workercmd.NewMain(logger, cfg, tc)
+		g.Add(
+			func() error {
+				if err := m.Run(ctx); err != nil {
+					return err
+				}
+				<-done
+				return nil
+			},
+			func(err error) {
+				if err := m.Close(); err != nil {
+					logger.Error(err, "Failed to close worker.")
+				}
+				close(done)
+			},
+		)
+	}
 
-	if err := m.Run(ctx); err != nil {
-		_ = m.Close()
-		os.Exit(1)
+	// API server.
+	{
+		srv := ais.NewAPIServer(ctx, tc, cfg.AIS.Listen, cfg.Temporal.TaskQueue)
+		g.Add(
+			func() error {
+				logger.Info("API server running", "listen", cfg.AIS.Listen)
+				return srv.ListenAndServe()
+			},
+			func(err error) {
+				ctx, cancel := context.WithTimeout(ctx, time.Second*5)
+				defer cancel()
+				_ = srv.Shutdown(ctx)
+			},
+		)
 	}
 
-	<-ctx.Done()
+	// Signal handler.
+	{
+		var (
+			cancelInterrupt = make(chan struct{})
+			ch              = make(chan os.Signal, 2)
+		)
+		defer close(ch)
 
-	if err := m.Close(); err != nil {
-		fmt.Fprintln(os.Stderr, err)
-		logger.Error(err, "Failed to close the application.")
+		g.Add(
+			func() error {
+				signal.Notify(ch, syscall.SIGINT, syscall.SIGTERM)
+
+				select {
+				case <-ch:
+				case <-cancelInterrupt:
+				}
+
+				return nil
+			},
+			func(err error) {
+				logger.Info("Quitting...")
+				close(cancelInterrupt)
+				cancel()
+				signal.Stop(ch)
+			},
+		)
+	}
+
+	err = g.Run()
+	if err != nil {
+		logger.Error(err, "Application failure.")
 		os.Exit(1)
 	}
+	logger.Info("Bye!")
```

Comment:
I just copied this `run.Group` implementation from Enduro to be able to run the worker and API server in the same process, there are some things I don't fully understand, but I hope we can remove the API soon and trigger this workflow from Enduro directly.